code/datums/elements/gravedigger.dm

```diff
@@ -0,0 +1,51 @@
+/**
+ * Gravedigger element. Allows for graves to be dug from certain tiles
+ */
+/datum/element/gravedigger
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+
+	/// How long it takes to dig a grave
+	var/dig_time = 8 SECONDS
+	/// A list of turf types that can be used to dig a grave.
+	var/static/list/turfs_to_consider = typecacheof(list(
+		/turf/open/misc/asteroid,
+		/turf/open/misc/dirt,
+		/turf/open/misc/grass,
+		/turf/open/misc/basalt,
+		/turf/open/misc/ashplanet,
+		/turf/open/misc/snow,
+		/turf/open/misc/sandy_dirt,
+	))
+
+/datum/element/gravedigger/Attach(datum/target, dig_time = 8 SECONDS)
+	. = ..()
+
+	if(!isobj(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.dig_time = dig_time
+	RegisterSignal(target, COMSIG_ITEM_INTERACTING_WITH_ATOM_SECONDARY, PROC_REF(dig_checks))
+
+/datum/element/gravedigger/Detach(datum/source, ...)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_INTERACTING_WITH_ATOM_SECONDARY)
+
+/datum/element/gravedigger/proc/dig_checks(datum/source, mob/living/user, atom/interacting_with, list/modifiers)
+	SIGNAL_HANDLER
+
+	if(!is_type_in_typecache(interacting_with, turfs_to_consider))
+		return NONE
+
+	if(locate(/obj/structure/closet/crate/grave) in interacting_with)
+		user.balloon_alert(user, "grave already present!")
+		return ITEM_INTERACT_BLOCKING
+
+	user.balloon_alert(user, "digging grave...")
+	playsound(interacting_with, 'sound/effects/shovel_dig.ogg', 50, TRUE)
+	INVOKE_ASYNC(src, PROC_REF(perform_digging), user, interacting_with, source)
+	return NONE
+
+/datum/element/gravedigger/proc/perform_digging(mob/user, atom/dig_area, obj/item/our_tool)
+	if(our_tool.use_tool(dig_area, user, 10 SECONDS))
```

Comment:
this is probably meant to use `dig_time` instead of `10 SECONDS`

Reply:
It's supposed to be 10 seconds, since the item's toolspeed is meant to be what determines how quickly the grave is dug. I must have forgotten to clear out that argument oopsies.